spot-client/src/spot-tv/app-state/actions.js

```diff
@@ -200,26 +202,49 @@ export function createSpotTVRemoteControlConnection({ pairingCode, retry }) {
          * @returns {Promise}
          */
         function onDisconnect(error) {
+            const isUnrecoverableError = remoteControlServer.isUnrecoverableRequestError(error);
+            const usingPermanentPairingCode = Boolean(pairingCode);
+
+            /**
+             * The connection is retried if explicit 'retry' flag has been set or if the connection succeeds initially.
+             * Even if both conditions are met, the connection will be dropped when using permanent pairing and
+             * the error is not recoverable.
+             *
+             * @type {boolean}
+             */
+            const willRetry
+                = (retry || initiallyConnected)
+                    && !(usingPermanentPairingCode && isUnrecoverableError);
```

Comment:
Would you be willing to break up these checks into other named variables? I'm having trouble keeping track of in what situations willRetry is true.

Reply:
yeah... although I'm not sure how to name them. The ideal was that first line is one variable and the second one another. Do you have any suggestions?

Reply:
`canRecoverConnection` for the second one?